### ISSUE TITLE
Handle subscriptions duration with different month lengths

### ISSUE THREAD

**Issue by nymous** [rezoleo/lea5#449]
When creating a subscription on the last day of May, which has 31 days, and lasts for 1 month (so until June 30th), Postgresql returns a duration of **0 months**.
This problem doesn't happen on May 30th (2024/05/30 -> 2024/06/30 = 1 month according to Postgresql).
See
```sql
select extract('months' from age('2024-06-30T12:00:00'::timestamp, '2024-05-31T12:00:00'::timestamp))
+---------+
| extract |
|---------|
| 0       |
+---------+

select extract('months' from age('2024-06-30T12:00:00'::timestamp, '2024-05-30T12:00:00'::timestamp))
+---------+
| extract |
|---------|
| 1       |
+---------+
```

To fix this, we first need to add a test in [`test/models/subscription_test.rb@SubscriptionTest#test_duration_should_give_months`](./test/models/subscription_test.rb#L66), with `travel_to Time.zone.local(2024, 5, 31)` and repeat the tests.

**Comment by benoitlx** [comment]
A possible workaround could be to truncate the start and end date of the subscription to the month before calculating the duration with the `age` postgre function. This can be implemented by adding `date_trunc` in a new migration :

```ruby
class FixDurationMonths < ActiveRecord::Migration[7.0]
  def change
    change_table :subscriptions, bulk: true do |t|
      t.remove :duration
      t.virtual :duration,
                type: :integer,
                as: "extract(year from age(date_trunc('months', end_at), date_trunc('months', start_at))) * 12 + extract(month from age(date_trunc('months', end_at), date_trunc('months', start_at)))",
                stored: true,
                comment: 'Duration in months'
    end
  end
end
```

With this quick fix, all current tests pass

**Comment by DioFun** [comment]
fixed by #450
